CBinding/Compiler/CCompiler.cs

```diff
@@ -33,75 +33,79 @@
 using System.IO;
 using System.Text;
 using System.CodeDom.Compiler;
+using System.Threading.Tasks;
 
 using MonoDevelop.Core;
 using MonoDevelop.Core.Execution;
 using MonoDevelop.Projects;
 using MonoDevelop.Ide;
+using System.Linq;
+
 
 namespace CBinding
 {
-	public abstract class CCompiler : ICompiler
+	public abstract class CCompiler : FolderBasedProject, ICompiler
```

Comment:
Because we have used `BaseDirectory.Combine` ,we can instead use `SolutionItem` there as well